src/basic/fd-util.c

```diff
@@ -208,10 +208,9 @@ static int get_max_fd(void) {
         return (int) (m - 1);
 }
 
-int close_all_fds(const int except[], size_t n_except) {
+int close_all_fds(int except[], size_t n_except) {
```

Comment:
hmpf. I really dislike this. we should make parameters constant. mucking around in input params is ugly as fuck.

at the absolute minimum this needs some explicit comment, because it breaks out usual style of having input params as constant, and not meddling with them on output

why not split this function in two: close_all_fds_no_malloc() that doesn't do the close_range() thing, and hence needs no sorting, and close_all_fds() as a wrapper around that which does. Then, from the signal handler call the former, fro the rest the latter.